oak_runtime/src/node/http/tests.rs

```diff
@@ -20,17 +20,71 @@ use oak_abi::{
     label::Label,
     proto::oak::application::{
         node_configuration::ConfigType, ApplicationConfiguration, HttpServerConfiguration,
-        NodeConfiguration,
+        NodeConfiguration, WebAssemblyConfiguration,
     },
 };
-use std::{fs, option::Option, thread::JoinHandle};
+use std::fs;
 
 static LOCAL_CA: &str = "../examples/certs/local/ca.pem";
 static GCP_CA: &str = "../examples/certs/gcp/ca.pem";
 
+struct OakNodeForTest {}
+
+impl crate::node::Node for OakNodeForTest {
+    fn node_type(&self) -> &'static str {
+        "test"
+    }
+    fn run(
+        self: Box<Self>,
+        runtime: RuntimeProxy,
+        handle: oak_abi::Handle,
+        _notify_receiver: oneshot::Receiver<()>,
+    ) {
+        // Get invocation message that contains the response_writer handle.
+        let invocation_receiver = Receiver::<HttpInvocation>::new(ReadHandle { handle });
+
+        while let Ok(invocation) = invocation_receiver.receive(&runtime) {
+            let request = invocation.receiver.unwrap().receive(&runtime);
+            info!("received request: {:?}", request);
+
+            let resp = HttpResponse {
+                body: vec![],
+                status: http::status::StatusCode::OK.as_u16() as i32,
+                headers: Some(HeaderMap {
+                    headers: hashmap! {},
+                }),
+            };
+            invocation
+                .sender
+                .expect("Empty sender on invocation.")
+                .send(resp, &runtime)
+                .unwrap();
+        }
+    }
+}
+
+struct TestNodeFactory {
```

Comment:
This needs way more documentation if we decide to keep it, but if we do keep it it may make sense to model this as a wrapper around the inner factory:

```rust
struct TestNodeFactory<T: NodeFactory> {
  inner: T,
}

enum TestNodeConfiguration {
  Test(Box<dyn Node>),
  Inner(NodeConfiguration),
}

impl NodeFactory<TestNodeConfiguration> for TestNodeFactory<T> {
  ...
}
```

or something like that.

I think this actually would allow to write more realistic tests overall, since the logic for creating the initial node graph would also be run from an actual (test) node, rather than by the runtime itself.

WDYT?

Reply:
I totally agree :)